src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -313,6 +313,33 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="ComputeCommitHash"
+          BeforeTargets="RepoBuild">
+
+        <!-- Figure out the repo commit. We try and accomodoate both minimal git repository metadata (https://github.com/dotnet/sourcelink/tree/main/docs#min
+mal-git-repository-metadata) or full git repo metadata. -->
+
+    <Exec Command="git rev-parse HEAD"
+          WorkingDirectory="$(RepoRoot)"
+          ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitRevParseHeadOutput" />
+    </Exec>
+
+    <ReadLinesFromFile File="$(RepoRoot)/.git/HEAD">
+      <Output TaskParameter="Lines" PropertyName="GitHeadContents" />
+    </ReadLinesFromFile>
```

Comment:
Executing `git rev-parse HEAD` and reading `.git/HEAD` is something sourcelink already does: https://github.com/dotnet/sourcelink/tree/main/docs#minimal-git-repository-metadata. Anyone know of an existing sourcelink task that we can re-use here to avoid this manual work?

Reply:
yes - check out `Microsoft.Build.Tasks.Git.LocateRepository`. You can see it in action in a 'normal' build during the `InitializeSourceControlInformationFromSourceControlManager` target, but its contents is the following:

```xml
  <Target Name="InitializeSourceControlInformationFromSourceControlManager">
    <!--
      Reports a warning if the given project doesn't belong to a repository under source control,
      unless the targets were implicily imported from an SDK without a package reference.
    -->
    <Microsoft.Build.Tasks.Git.LocateRepository 
      Path="$(MSBuildProjectDirectory)"
      RemoteName="$(GitRepositoryRemoteName)"
      ConfigurationScope="$(GitRepositoryConfigurationScope)"
      NoWarnOnMissingInfo="$(PkgMicrosoft_Build_Tasks_Git.Equals(''))">

      <Output TaskParameter="RepositoryId" PropertyName="_GitRepositoryId" />
      <Output TaskParameter="Url" PropertyName="ScmRepositoryUrl" />
      <Output TaskParameter="Roots" ItemName="SourceRoot" />
      <Output TaskParameter="RevisionId" PropertyName="SourceRevisionId" Condition="'$(SourceRevisionId)' == ''" />
    </Microsoft.Build.Tasks.Git.LocateRepository>

    <PropertyGroup>
      <RepositoryType Condition="'$(RepositoryType)' == ''">git</RepositoryType>
    </PropertyGroup>
  </Target>
```

If you'd rather re-use already-computed information, you could see if the `@(SourceRoot)` items have been populated yet. SourceLink will generate and add metadata to those, so you could get the same information by finding the `SourceRoot` item with a `SourceControl` metadata of `git` and then reading the `RevisionId` metadata from that item:

![image](https://github.com/dotnet/installer/assets/573979/c4ff6d12-8bf2-4183-9408-cdcd5110bcd7)